bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/handler/AVMFritzButtonHandler.java

```diff
@@ -89,11 +95,44 @@ public void onDeviceUpdated(ThingUID thingUID, AVMFritzBaseModel device) {
         }
     }
 
+    @Override
+    protected void updateTemperatureSensor(@Nullable TemperatureModel temperatureModel) {
+        if (temperatureModel != null) {
+            String channelId = (DECT440_THING_TYPE.equals(thing.getThingTypeUID())
+                    ? CHANNEL_GROUP_SENSORS + ChannelUID.CHANNEL_GROUP_SEPARATOR
+                    : "") + CHANNEL_TEMPERATURE;
+            updateThingChannelState(channelId, new QuantityType<>(temperatureModel.getCelsius(), SIUnits.CELSIUS));
+            updateThingChannelConfiguration(channelId, CONFIG_CHANNEL_TEMP_OFFSET, temperatureModel.getOffset());
+        }
+    }
+
     @Override
     protected void updateHumiditySensor(@Nullable HumidityModel humidityModel) {
         if (humidityModel != null) {
-            updateThingChannelState(CHANNEL_GROUP_SENSORS + ChannelUID.CHANNEL_GROUP_SEPARATOR + CHANNEL_HUMIDITY,
-                    new QuantityType<>(humidityModel.getRelativeHumidity(), Units.PERCENT));
+            String channelId = (DECT440_THING_TYPE.equals(thing.getThingTypeUID())
+                    ? CHANNEL_GROUP_SENSORS + ChannelUID.CHANNEL_GROUP_SEPARATOR
+                    : "") + CHANNEL_HUMIDITY;
+            updateThingChannelState(channelId, new QuantityType<>(humidityModel.getRelativeHumidity(), Units.PERCENT));
+        }
+    }
+
+    @Override
+    protected void updateBattery(BatteryModel batteryModel) {
+        String batteryLevelChannelId = (DECT440_THING_TYPE.equals(thing.getThingTypeUID())
+                ? CHANNEL_GROUP_DEVICE + ChannelUID.CHANNEL_GROUP_SEPARATOR
+                : "") + CHANNEL_BATTERY;
+        BigDecimal batteryLevel = batteryModel.getBattery();
+        updateThingChannelState(batteryLevelChannelId,
+                batteryLevel == null ? UnDefType.UNDEF : new DecimalType(batteryLevel));
+        String lowBatteryChannelId = (DECT440_THING_TYPE.equals(thing.getThingTypeUID())
+                ? CHANNEL_GROUP_DEVICE + ChannelUID.CHANNEL_GROUP_SEPARATOR
+                : "") + CHANNEL_BATTERY_LOW;
+        BigDecimal lowBattery = batteryModel.getBatterylow();
+        if (lowBattery == null) {
+            updateThingChannelState(lowBatteryChannelId, UnDefType.UNDEF);
+        } else {
+            updateThingChannelState(lowBatteryChannelId,
+                    BatteryModel.BATTERY_ON.equals(lowBattery) ? OnOffType.ON : OnOffType.OFF);
```

Comment:
Is this supposed to be an equality and not a comparison?

Reply:
It is a comparison.

Reply:
Ah, after looking at the code I see why `.equals` is used.